save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/ExecutionController.kt

```diff
@@ -98,30 +103,46 @@ class ExecutionController(private val executionService: ExecutionService,
                 .map { it.toDto() }
 
     /**
-     * @param name
+     * @param projectName
      * @param authentication
      * @param organizationName
+     * @param filters
      * @return list of execution dtos
      */
-    @GetMapping(path = ["/api/$v1/executionDtoList"])
+    @Suppress("PARAMETER_NAME_IN_OUTER_LAMBDA")
+    @PostMapping(path = ["/api/$v1/executionDtoList"])
     fun getExecutionByProject(
-        @RequestParam name: String,
+        @RequestParam projectName: String,
         @RequestParam organizationName: String,
+        @RequestBody(required = false) filters: ExecutionFilters?,
         authentication: Authentication,
     ): Mono<List<ExecutionDto>> = organizationService.findByName(organizationName)
         .toMono()
         .switchIfEmptyToNotFound {
             "Organization with name [$organizationName] was not found."
         }
-        .flatMap { organization ->
-            projectService.findWithPermissionByNameAndOrganization(authentication, name, organization.name, Permission.READ).map {
-                executionService.getExecutionByNameAndOrganization(name, organization).filter {
-                    it.type != TestingType.CONTEST_MODE
-                }.map {
-                    it.toDto()
-                }
-                    .reversed()
+        .zipWhen { organization ->
+            projectService.findWithPermissionByNameAndOrganization(authentication, projectName, organization.name, Permission.READ)
+        }
+        .map { (organization, _) ->
+            filters?.let {
+                val startTime = createLocalDateTime(filters.startTime)
+                val endTime = filters.endTime?.let { createLocalDateTime(it) } ?: getEndLocalDateTime(filters.startTime)
```

Comment:
I'd suggest to put this logic into `getEndLocalDateTime` completely

Reply:
Removed this logic